### PR TITLE
Fix linting errors

### PR DESCRIPTION
**Pull request by aykxt**
This PR fixes some linting errors and removes unnecessary mutable borrows.

### CODE REVIEW

File: lib/grammers-client/src/utils.rs
Comment (Lonami) [inline lib/grammers-client/src/utils.rs:32]: This is semantically different, but I guess in this case it's always going to be `Ok`.

File: lib/grammers-mtsender/src/lib.rs
Comment (Lonami) [inline lib/grammers-mtsender/src/lib.rs:63]: Also semantically different.

File: lib/grammers-mtsender/src/lib.rs
Comment (Lonami) [inline lib/grammers-mtsender/src/lib.rs:473]: I don't think we can `unwrap` this one. The user may have dropped the oneshot receiver by the time a response from Telegram arrives. In this case we actually want to ignore. If clippy is complaining, perhaps we could `debug!()` `if .is_err()`.

File: lib/grammers-session/src/lib.rs
Comment (Lonami) [inline lib/grammers-session/src/lib.rs:123]: Why?

**Review by Lonami — CHANGES_REQUESTED** [review]
I would have preferred if the `mut` changes were a separate PR or, at least, a separate commit. Same for the semantic changes regarding `drop`.

That said, after addressing the `drop` changes, I'll merge it regardless if separating the `mut` stuff into another PR is too much effort.

File: lib/grammers-session/src/lib.rs
Comment (aykxt) [reply lib/grammers-session/src/lib.rs:123]: I think it's obvious which address is meant, also server_addr was aliased to addr later in the code. Should I rename it back to server_addr?

File: lib/grammers-mtsender/src/lib.rs
Comment (aykxt) [reply lib/grammers-mtsender/src/lib.rs:473]: so leave it to another PR and revert or should I add the debug log?

**Comment by aykxt** [comment]
> That said, after addressing the `drop` changes, I'll merge it regardless if separating the `mut` stuff into another PR is too much effort.

I think it's not worth separating the `mut` changes into a different PR (they are also some kind of linting errors, aren't they?).

File: lib/grammers-session/src/lib.rs
Comment (Lonami) [reply lib/grammers-session/src/lib.rs:123]: Nah it's fine, I just wondered.

File: lib/grammers-mtsender/src/lib.rs
Comment (Lonami) [reply lib/grammers-mtsender/src/lib.rs:473]: Please remove this change from this one and then we can send a separate one with a better "fix".

**Comment by Lonami** [comment]
> [...] the `mut` changes [...] (they are also some kind of linting errors, aren't they?).

No, they're also semantic changes, not just pure stylistic changes from the clippy lints. However, if it's too much effort to separate them, I can still merge as is.

File: lib/grammers-mtsender/src/lib.rs
Comment (aykxt) [reply lib/grammers-mtsender/src/lib.rs]: done

**Comment by Lonami** [comment]
Thanks.

**Pull request merged**
